sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/RecoveryCodePDF.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.sync.impl
+
+import android.content.Context
+import android.graphics.pdf.PdfDocument
+import android.graphics.pdf.PdfDocument.PageInfo.Builder
+import android.os.Environment
+import android.view.LayoutInflater
+import android.view.View
+import com.duckduckgo.app.global.DispatcherProvider
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.mobile.android.ui.view.toPx
+import com.duckduckgo.sync.impl.databinding.ViewRecoveryCodeBinding
+import com.squareup.anvil.annotations.ContributesBinding
+import java.io.File
+import java.io.FileOutputStream
+import javax.inject.*
+import kotlinx.coroutines.withContext
+
+interface RecoveryCodePDF {
+    suspend fun generateAndStoreRecoveryCodePDF(
```

Comment:
this doesn’t need to be `suspend`, adding a `@WorkerThread` annotation would be better. The `ViewModel` can/should deal with threading